src/Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -1,2180 +1,2682 @@
+            }
+
+            var methodDefinitionHandle = AddMethodDefinition(
+                methodSymbol.Name,
+                parameters,
+                new Symbol(methodSymbol.ReturnType),
+                IsStaticNode(node),
+                node.Parent is InterfaceDeclarationSyntax);
+            currentTypeDeclaration.AddMethod(methodSymbol, methodSymbol.Name, methodDefinitionHandle);
+        }
+
+        public override void VisitFieldDeclaration(FieldDeclarationSyntax node)
+        {
+            if (!IsPublicNode(node) || node.Parent is not StructDeclarationSyntax)
+            {
+                return;
+            }
+
+            base.VisitFieldDeclaration(node);
+
+            var symbol = Model.GetTypeInfo(node.Declaration.Type).Type;
+            var fieldSignature = new BlobBuilder();
+            var encoder = new BlobEncoder(fieldSignature);
+
+            if (IsEncodableAsSpecialType(symbol.SpecialType))
+            {
+                EncodeSpecialType(symbol.SpecialType, encoder.FieldSignature());
+            }
+            else
+            {
+                EncodeSymbol(new Symbol(symbol), encoder.FieldSignature());
+            }
+
+            foreach (var variable in node.Declaration.Variables)
+            {
+                var fieldSymbol = Model.GetDeclaredSymbol(variable);
+                var fieldDefinitionHandle = metadataBuilder.AddFieldDefinition(
+                    FieldAttributes.Public,
+                    metadataBuilder.GetOrAddString(variable.Identifier.Text),
+                    metadataBuilder.GetOrAddBlob(fieldSignature));
+                currentTypeDeclaration.AddField(fieldSymbol, fieldDefinitionHandle);
+            }
+        }
+
+        public void AddPropertyDefinition(
+            string propertyName,
+            Symbol type,
+            ISymbol symbol,
+            bool hasSetMethod,
+            bool isInterfaceParent)
+        {
+            Logger.Log("defining property " + propertyName);
+
+            var propertySignature = new BlobBuilder();
+            new BlobEncoder(propertySignature)
+                .PropertySignature(true)
+                .Parameters(
+                    0,
+                    returnType => EncodeReturnType(type, returnType),
+                    parameters => { }
+                );
+
+            var propertyDefinitonHandle = metadataBuilder.AddProperty(
+                PropertyAttributes.None,
+                metadataBuilder.GetOrAddString(propertyName),
+                metadataBuilder.GetOrAddBlob(propertySignature));
+            currentTypeDeclaration.AddProperty(symbol, propertyDefinitonHandle);
+
+            if (hasSetMethod)
+            {
+                string setMethodName = "put_" + propertyName;
+                var setMethod = AddMethodDefinition(
+                    setMethodName,
+                    new Parameter[] { new Parameter(type, "value", ParameterAttributes.In) },

```

Comment:
Do we need to consider naming conflicts with user params?

Reply:
I am hoping naming conflicts will be caught by diagnostics before we even get here.  We can in the future maybe support overriding the name if allowed by WinRT as an enhancement.